crates/red_knot_python_semantic/src/types.rs

```diff
@@ -34,18 +36,15 @@ mod mro;
 mod narrow;
 mod unpacker;
 
-pub fn check_types(db: &dyn Db, file: File) -> TypeCheckDiagnostics {
+#[salsa::tracked]
+pub fn check_types(db: &dyn Db, file: File) {
```

Comment:
This has to be a salsa query so that the mdtest framework can get the diagnostics (accumulator require a query)